.github/workflows/release.yml

```diff
@@ -0,0 +1,38 @@
+name: Release
+
+on:
+  push:
+    tags: [ '**' ]
+  workflow_dispatch:
+
+concurrency: release-${{ github.ref }}
+
+env:
+  IMAGE_NAME: hri-flink-pipeline-core
+
+jobs:
+  buildAndRelease:
+    name: Gradle Build and Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set branch name
+        uses: nelonoel/branch-name@v1.0.1
+
+      - uses: actions/checkout@v2
+
+      - name: Set up adoptopenjdk8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Gradle build
+        run: |
+          chmod +x ./gradlew
+          ./gradlew clean build
+
+      - name: Gradle publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew publish --stacktrace
```

Comment:
Maybe I'm missing something, but this workflow looks the same as `ci-workflow.yml`.  Could we combine them? I think the key would be correctly determining the artifact ID in the Gradle build.  Branches would need the `-SNAPSHOT` suffix, while tags are supposed to match the version.

Reply:
I modeled the two workflows after the ones in `hri-mgmt-api`.  Those are also similar but they have different vulnerability checks and scans.  I assumed that when Aram would add in those checks and scans to this branch that the functionality would diverge and they would be more different in the way the two are in `hri-mgmt-api`.  I can combine them instead.

Reply:
> I modeled the two workflows after the ones in `hri-mgmt-api`. Those are also similar but they have different vulnerability checks and scans. I assumed that when Aram would add in those checks and scans to this branch that the functionality would diverge and they would be more different in the way the two are in `hri-mgmt-api`. I can combine them instead.

Yeah, @alisakotliarova makes a good point here. The OpenSource/Vulnerability scan will probably only happen in another (2nd) `pull_request.yml` workflow (as per this comment --> https://github.com/Alvearie/hri-flink-pipeline-core/pull/3/files#r716981860). 

So we should have a separate `pull_request.yml` workflow I modeled on https://github.com/Alvearie/hri-mgmt-api/blob/develop/.github/workflows/pull_request.yml, @David-N-Perkins. And, then the Vulnerability scan call will get put into that one.